python/ray/autoscaler/_private/vsphere/pyvmomi_sdk_provider.py

```diff
@@ -25,7 +25,30 @@ def __init__(self, server, user, password, session_type: Constants.SessionType):
         atexit.register(Disconnect, smart_connect_obj)
         self.pyvmomi_sdk_client = smart_connect_obj.content
 
-    def get_pyvmomi_obj(self, vimtype, name):
+    def get_pyvmomi_obj_by_moid(self, vimtype, moid):
+        obj = None
+        if self.pyvmomi_sdk_client is None:
+            raise ValueError("Must init pyvmomi_sdk_client first.")
+
+        container = self.pyvmomi_sdk_client.viewManager.CreateContainerView(
+            self.pyvmomi_sdk_client.rootFolder, vimtype, True
+        )
+
+        for c in container.view:
+            if moid:
+                if moid in str(c):
+                    obj = c
+                    break
+            else:
+                obj = c
+                break
+        if not obj:
+            raise RuntimeError(
+                f"Unexpected: cannot find vSphere object {vimtype} with moid: {moid}"
+            )
+        return obj
+
+    def get_pyvmomi_obj_by_name(self, vimtype, name):
```

Comment:
Can you add comments to this function?

Reply:
It already has comment in the function. 
![image](https://github.com/ray-project/ray/assets/85480625/5948dd98-f8d1-47b3-b6fa-0cf24fe84ede)